### PR TITLE
feat: content type metadata support

### PR DESCRIPTION
**Pull request by IBratWurst**
## Summary

Introduce metadata to the tool and support for annotations management

## Description

Enable the tool to tolerate existing annotations on content type payloads.
Add functionalities to edit annotations assignments for content types and fields

### CODE REVIEW

**Review by marcolink — APPROVED** [review]

File: src/lib/intent/interface-example.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable unicorn/filename-case */
```
Comment (anho) [inline src/lib/intent/interface-example.js:1]: Is this file a leftover?

File: src/lib/interfaces/annotation.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  data: Record<string, any>
+}
+
+export type AnnotationAssignment = {
+  annotationId: string
+  target: AssignmentTarget
+}
+
+/**
+ * This data structure does not include all the properties defined in CEP-0125
```
Comment (anho) [inline src/lib/interfaces/annotation.ts:20]: I think we can remove the notion of a CEP from a public repo.

**Review by anho — CHANGES_REQUESTED** [review]
Awesome. Just minor stuff left.

**Comment by IBratWurst** [comment]
> Awesome. Just minor stuff left.

🙏🏼  Thanks for spotting those

File: src/lib/action/field-rename.ts [hunk 3]
```diff
@@ -29,11 +29,15 @@ class FieldRenameAction extends EntityAction {
     const fields = ct.fields
     const field = fields.getField(this.fieldId)
 
+    const fieldAnnotations = ct.getFieldAnnotations(this.fieldId)
     fields.deleteField(this.fieldId)
     field.id = this.props.newId
     delete field.newId
 
     fields.setField(this.props.newId, field)
+    if (fieldAnnotations) {
```
Comment (veu) [inline src/lib/action/field-rename.ts:38]: Can’t hurt to add a test for this as well. E.g. so it doesn’t get overlooked in a refactoring.

Comment (IBratWurst) [reply src/lib/action/field-rename.ts:38]: Is this it https://github.com/contentful/contentful-migration/pull/1060/files#diff-72b02b4b4284f14f9bfdfe3de28700e0469eb4d53066c70719c59a89846f3040 ? :)

Comment (veu) [reply src/lib/action/field-rename.ts:38]: Yes, that works. I had only seen the examples up to this point.

**Review by anho — APPROVED** [review]

**Pull request merged**